packages/sqflite/tizen/src/sqflite_plugin.cc

```diff
@@ -0,0 +1,785 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sqflite_plugin.h"
+
+#include <app_common.h>
+#include <dlog.h>
+#include <flutter/event_channel.h>
+#include <flutter/event_sink.h>
+#include <flutter/event_stream_handler_functions.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar.h>
+#include <flutter/standard_method_codec.h>
+#include <storage.h>
+
+#include <filesystem>
+#include <list>
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "constants.h"
+#include "database_manager.h"
+#include "errors.h"
+#include "log.h"
+
+template <typename T>
+bool GetValueFromEncodableMap(flutter::EncodableMap &map, std::string key,
+                              T &out) {
+  auto iter = map.find(flutter::EncodableValue(key));
+  if (iter != map.end() && !iter->second.IsNull()) {
+    if (auto pval = std::get_if<T>(&iter->second)) {
+      out = *pval;
+      return true;
+    }
+  }
+  return false;
+}
+
+struct DBResultVisitor {
+  flutter::EncodableValue operator()(int64_t val) {
+    return flutter::EncodableValue(val);
+  };
+  flutter::EncodableValue operator()(std::string val) {
+    return flutter::EncodableValue(val);
+  };
+  flutter::EncodableValue operator()(double val) {
+    return flutter::EncodableValue(val);
+  };
+  flutter::EncodableValue operator()(std::vector<uint8_t> val) {
+    return flutter::EncodableValue(val);
+  };
+  flutter::EncodableValue operator()(std::nullptr_t val) {
+    return flutter::EncodableValue();
+  };
+};
+
+class SqflitePlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrar *registrar) {
+    auto channel =
+        std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
+            registrar->messenger(), sqflite_constants::kPluginKey,
+            &flutter::StandardMethodCodec::GetInstance());
+
+    auto plugin = std::make_unique<SqflitePlugin>(registrar);
+
+    channel->SetMethodCallHandler(
+        [plugin_pointer = plugin.get()](const auto &call, auto result) {
+          plugin_pointer->HandleMethodCall(call, std::move(result));
+        });
+
+    registrar->AddPlugin(std::move(plugin));
+  }
+  SqflitePlugin(flutter::PluginRegistrar *registrar) : registrar_(registrar) {}
+
+  virtual ~SqflitePlugin() {}
+
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    LOG_DEBUG("HandleMethodCall: %s", method_call.method_name().c_str());
+    const std::string method_name = method_call.method_name();
+    if (method_name == sqflite_constants::kMethodOpenDatabase) {
```

Comment:
It's up to your preference but I personally prefer
```suggestion
    if (method_name == "openDatabase") {
```
in terms of readability.

Reply:
I'll keep the constants since I think is a good practice to preserve fixed strings in constant variables in case we need to use them somewhere else.